src/feature/control/control_getinfo.c

```diff
@@ -326,46 +326,33 @@ getinfo_helper_current_time(control_connection_t *control_conn,
 }
 
 /**
- * Switch between microdesc vs networkstatus descriptor dumps
- * Assumes that question would be either:
- * - dir/status-vote/current/consensus-microdesc
- * or
- * - dir/status-vote/current/consensus
+ * Helps to access directory information for a given
+ * type of consensus.
+ *
+ * returns: int
+ *   -1 in cases of an error
+ *   0 otherwise
  */
 STATIC int
-getinfo_helper_current_consensus(const char* question,
-                          char** answer,
-                          const char** errmsg)
+getinfo_helper_current_consensus(const char* consensus_type,
+                                 char** answer, const char** errmsg)
 {
-  // Ensures question contains a request for
-  // ns or microdesc consensus
-  if (
-      strcmp(question, "dir/status-vote/current/consensus") &&
-      strcmp(question, "dir/status-vote/current/consensus-microdesc")) {
-    return 0;
-  }
-
-  const char* consensus_type = !strcmp(
-     question, "dir/status-vote/current/consensus-microdesc"
-  ) ? "microdesc" : "ns";
 
-  if (we_want_to_fetch_flavor(get_options(), FLAV_NS)) {
-    const cached_dir_t *consensus = dirserv_get_consensus(consensus_type);
-    if (consensus) {
-      *answer = tor_strdup(consensus->dir);
-    }
```

Comment:
Are we sure we don't need the `we_want_to_fetch_flavor()` call here anymore? What purpose was it serving in the past?

Reply:
I checked `we_want_to_fetch_flavor()` function and I saw that it returns a list of consensus flavors. since we are passing that ourselves, I thought about removing it.

Reply:
I think the we_want_to_fetch_flavor() check is used to make sure that the flavour is in RAM or on disk.